File: Professional Image Modifier 5/[VB]-Professional Image Modifier 5/VB/ImageFunctions/Forms/FrmFileLoader.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports ImageFunctions.Controls
Imports WeifenLuo.WinFormsUI.Docking

Namespace ImageFunctions.Forms
	Partial Public Class FrmFileLoader
		Inherits DockContent

		Public Delegate Sub ThumbnailSelectedHandler(ByVal PathToMedia As String)
		Public Event ThumbnailSelected As ThumbnailSelectedHandler


		Public Sub New()
			InitializeComponent()
		End Sub

		Protected Overrides Function GetPersistString() As String
			Return Me.Text
		End Function

		Friend Sub Add(ByVal thumbnail As ThumbnailControl)
			AddHandler thumbnail.ThumbnailSelected, AddressOf thumbnail_ThumbnailSelected
			ThumbnailLayoutPanel.Controls.Add(thumbnail)

		End Sub

		Private Sub thumbnail_ThumbnailSelected(ByVal PathToMedia As String)
			RaiseEvent ThumbnailSelected(PathToMedia)
		End Sub



	End Class
End Namespace

File: Professional Image Modifier 5/[C#]-Professional Image Modifier 5/C#/ImageFunctions/Forms/FrmFileLoader.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using ImageFunctions.Controls;
using WeifenLuo.WinFormsUI.Docking;

namespace ImageFunctions.Forms
{
    public partial class FrmFileLoader : DockContent
    {
        public delegate void ThumbnailSelectedHandler(string PathToMedia);
        public event ThumbnailSelectedHandler ThumbnailSelected;


        public FrmFileLoader()
        {
            InitializeComponent();
        }

        protected override string GetPersistString()
        {
            return this.Text;
        }

        internal void Add(ThumbnailControl thumbnail)
        {
            thumbnail.ThumbnailSelected += thumbnail_ThumbnailSelected;
            ThumbnailLayoutPanel.Controls.Add(thumbnail);

        }

        void thumbnail_ThumbnailSelected(string PathToMedia)
        {
            ThumbnailSelected(PathToMedia);
        }



    }
}
